Userland/Libraries/LibJS/Tests/syntax/coalesce-logic-expression-mixing.js

```diff
@@ -20,6 +20,19 @@ test("mixing coalescing and logical operators with parens", () => {
     expect("if (0) a || (b * c) ?? d").not.toEval();
 });
 
+test("mixing coalescing and logical operators in ternary expressions", () => {
+    expect("0 || 0 ? 0 : 0 ?? 0").toEval();
+    expect("0 ?? 0 ? 0 : 0 || 0").toEval();
+    expect("0 ? 0 || 0 : 0 ?? 0").toEval();
+    expect("0 ? 0 ?? 0 : 0 || 0").toEval();
+    expect("0 && 0 ? 0 ?? 0 : 0 || 0").toEval();
+    expect("0 ?? 0 ? 0 && 0 : 0 || 0").toEval();
+    expect("0 ?? 0 ? 0 || 0 : 0 && 0").toEval();
+    expect("0 || 0 ? 0 ?? 0 : 0 && 0").toEval();
+    expect("0 && 0 ? 0 || 0 : 0 ?? 0").toEval();
+    expect("0 || 0 ? 0 && 0 : 0 ?? 0").toEval();
```

Comment:
Were we only too strict?
I.e. there are no new cases we shouldn't eval?

Reply:
"mixing coalescing and logical operators isn't allowed" above already covers various cases that are not allowed, and I couldn't think of anything else